backoff/backoff.go

```diff
@@ -0,0 +1,71 @@
+package backoff
+
+import (
+	"time"
+)
+
+type RandomGenerator interface {
+	Int63n(n int64) int64
+}
+
+type Config struct {
+	Base                 time.Duration
+	Max                  time.Duration
+	BackOffResetDuration time.Duration
+	Jitter               Jitter
+}
+
+var DefaultConfig = &Config{
+	Base:                 time.Second,
+	Max:                  time.Second * 30,
+	BackOffResetDuration: time.Second * 60,
+	Jitter:               FullJitter,
+}
+
+// Backoff The backoff is calculated by min(Max, rand(0, Base*(2^retries))).
+type Backoff struct {
```

Comment:
A more descriptive documentation is needed.
What this component is for. 
Example usage etc.

Reply:
Added more comments.